src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -1324,15 +1316,39 @@ .. chosenResolvedItem.Suppressions
                         continue;
                     }
 
+                    // See if a dependency with the same version and suppressions has already been chosen
+                    if (resolvedDependencyGraphItems.TryGetValue(childLibraryDependencyIndex, out ResolvedDependencyGraphItem? childResolvedDependencyGraphItem)
+                        && childResolvedDependencyGraphItem.LibraryRangeIndex == childLibraryRangeIndex
+                        && childResolvedDependencyGraphItem.Suppressions.Count == 1
+                        && childResolvedDependencyGraphItem.Suppressions[0].Count == 0)
+                    {
+                        if (!childResolvedDependencyGraphItem.IsRootPackageReference)
+                        {
+                            childResolvedDependencyGraphItem.Parents ??= new HashSet<LibraryRangeIndex>();
```

Comment:
It looks like a change got pushed while I was reviewing so I'm leaving a comment again. It would be better to have the instantiation of the Parents HashSet property be encapsulated in its own object rather than by consumers of the object.